### PR TITLE
test: Add e2e test for `Mock` tag

### LINKED ISSUE #425: Add e2e test for `Mock` tag
We should check the `mock` tag exists in mock mode

### PR DESCRIPTION
**Pull request by shriram2301**
resolves #425

### CODE REVIEW

**Comment by shriram2301** [comment]
-[ ] test if `Mock Network` tag is present or absent based on whether node env is `test` or `dev` or `prod`

**Comment by joshuef** [comment]
Also `peruse/spec` is the wrong place for these tests, they are SAFE specific, so should live in the extensions.

File: app/constants.js [hunk 1]
```diff
@@ -234,7 +234,8 @@ export const CLASSES = {
     SETTINGS_MENU__HISTORY       : 'js-settingsMenu_history',
     SETTINGS_MENU__TOGGLE        : 'js-settingsMenu_toggle',
     SETTINGS_MENU__TOGGLE_BUTTON : 'js-settingsMenu_toggleButton',
-    SETTINGS_MENU__TOGGLE_TEXT   : 'js-settingsMenu_toggleText'
+    SETTINGS_MENU__TOGGLE_TEXT   : 'js-settingsMenu_toggleText',
+    MOCK_TAG                     : 'js-MockTag'
```
Comment (joshuef) [inline app/constants.js]: Please keep this inline with other class names (ie. camel case)

Comment (joshuef) [reply app/constants.js]: and perhaps provide a more descriptive name

File: __e2e__/peruse.spec.js [hunk 2]
```diff
@@ -340,6 +340,52 @@ describe( 'main window', () =>
...
+        await delay( 4500 );
+        await newTab( app );
+        await delay( 2500 );
+        if ( nodeEnv === 'dev')
+        {
+            expect( await client.waitForExist( BROWSER_UI.MOCK_TAG, WAIT_FOR_EXIST_TIMEOUT ) ).toBeTruthy();
+        }
+        if ( nodeEnv === 'prod')
+        {
+            expect( await client.isExisting( BROWSER_UI.MOCK_TAG, WAIT_FOR_EXIST_TIMEOUT, true ) ).toBeFalsy();
```
Comment (joshuef) [inline __e2e__/peruse.spec.js]: Just running this text, in mock passes. Ie: this test doesn't appear to actually check what you want...

Why is it different to the above test? (ln 352?)

Comment (joshuef) [reply __e2e__/peruse.spec.js]: These should really be two separate tests (running only if in that environment)

File: app/constants.js [hunk 3]
```diff
@@ -38,6 +38,11 @@ if ( isRunningSpectronTestProcess && !isRunningSpectronTestProcessingPackagedApp
     shouldRunMockNetwork = true;
 }
 
+if ( isRunningSpectronTestProcess && !isRunningSpectronTestProcessingPackagedApp && allPassedArgs.includes( '--live' ) )
```
[the remark this answers is not in the document]
Comment (shriram2301) [reply app/constants.js]: Yes, it is possible,I tried running it locally just now with this suggestion.I'll remove this `if` and reorder the `if` statements.

File: app/extensions/safe/test/e2e/safe-mock-live.spec.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    } );
+
+    it( 'checks for Mock Network Tag to be absent when --live parameter is passed', async () => 
+    {
+        expect.assertions ( 1 );
+        const { client } = app;
+        await delay( 4500 );
+        await newTab( app );
+        await delay( 2500 );
+        expect( await client.isExisting( BROWSER_UI.MOCK_TAG, WAIT_FOR_EXIST_TIMEOUT, true ) ).toBeFalsy();
```
[the remark this answers is not in the document]
Comment (shriram2301) [reply app/extensions/safe/test/e2e/safe-mock-live.spec.js]: oh ok I get it. The mock and live ones are disparate. I will change this.

**Pull request merged**